src/element-service.js

```diff
@@ -166,85 +172,66 @@ function assertService(service, id, extension) {
   ));
 }
 
-/**
- * Waits for body to be present then verifies that an extension script is
- * present in head for installation.
- * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
- * @param {string} extensionId
- * @return {!Promise<boolean>}
- */
-export function isExtensionScriptInNode(ampdoc, extensionId) {
-  return ampdoc.waitForBodyOpen().then(() => {
-    return extensionScriptInNode(ampdoc.getHeadNode(), extensionId);
-  });
-}
-
 /**
  * Verifies that an extension script is present in head for
  * installation.
- * @param {HTMLHeadElement|Element|ShadowRoot} head
+ * @param {!Window} win
  * @param {string} id
+ * @param {string} version
  * @return {boolean}
  * @private
  */
-function extensionScriptInNode(head, id) {
-  return extensionScriptsInNode(head).some(
-    ({extensionId}) => id == extensionId
+function extensionScriptInNode(win, id, version) {
+  return extensionScriptsInNode(win.document.head).some(
+    ({extensionId, extensionVersion}) =>
+      id == extensionId && version == extensionVersion
   );
 }
 
-/**
- * Waits for an extension if its script is present
- * @param {!Window} win
- * @param {string} extension
- * @param {HTMLHeadElement|Element|ShadowRoot} head
- * @return {!Promise}
- * @private
- */
-function waitForExtensionIfPresent(win, extension, head) {
-  /**
-   * If there is an extension script wait for it to load before trying
-   * to get the service. Prevents a race condition when everything but
-   * the extensions is in cache. If there is no script then it's either
-   * not present, or the service was defined by a test. In those cases
-   * we don't wait around for an extension that does not exist.
-   */
-
-  // TODO(jpettitt) investigate registerExtension to short circuit
-  // the dom call in extensionScriptsInNode()
-  if (!extensionScriptInNode(head, extension)) {
-    return Promise.resolve();
-  }
-
-  const extensions = getService(win, 'extensions');
-  return /** @type {!Promise<?Object>} */ (extensions.waitForExtension(
-    win,
-    extension
-  ));
-}
-
 /**
  * Returns the promise for service with `id` on the given window if available.
  * Otherwise, resolves with null (service was not registered).
  * @param {!Window} win
  * @param {string} id
  * @param {string} extension
+ * @param {string} version
  * @param {boolean=} opt_element
  * @return {!Promise<Object>}
  * @private
  */
-function getElementServicePromiseOrNull(win, id, extension, opt_element) {
+function getElementServicePromiseOrNull(
+  win,
+  id,
+  extension,
+  version,
+  opt_element
+) {
   return dom
     .waitForBodyOpenPromise(win.document)
```

Comment:
Shouldn't this switch to `whenExtensionsKnown`, too?

Reply:
`whenExtensionsKnown` are only available on an ampdoc, not the global window. This was part of a reason why we tried to remove all global services in extensions. Unfortunately, the new ones crept in (mostly from `amp-story`), but it'd be hard to refactor them all in one go.